### PR TITLE
feat: softdelete rule model

### PR DESCRIPTION
**Pull request by trbtm**
I would like to have the option for having soft delete rules in the database to be able to possibly track fields like

- deleted_by, deleted_at
- updated_by, updated_at

etc.

### CODE REVIEW

**Comment by leeqvip** [comment]
@trbtm  Hi,
1. Can the database operations be separated? The Adapter only needs to call the corresponding query or modification method.
2. Regarding soft deletion, some fields are marked as `true` or `false`, while others set the `deleted_at` time.
3. The updated_at you mentioned is not shown here.

**Comment by trbtm** [comment]
@leeqvip Hi,

I added an example demonstrating my approach. To your points:

1. I think that would be difficult since deleted rules need to be filtered when the adapter is querying the database e.g. in `Adapter.remove_policies` you want to make sure that only non deleted rules get affected
2. I think this gets clear when you look the the example I provided
3. Same as above.

Best
Thore

File: casbin_sqlalchemy_adapter/adapter.py [hunk 1]
```diff
@@ -56,15 +56,27 @@ class Filter:
 class Adapter(persist.Adapter, persist.adapters.UpdateAdapter):
     """the interface for Casbin adapters."""
 
-    def __init__(self, engine, db_class=None, filtered=False, create_all_models=True):
+    def __init__(
+        self,
+        engine,
+        db_class=None,
+        db_class_softdelete_attribute=None,
```
Comment (leeqvip) [inline casbin_sqlalchemy_adapter/adapter.py:63]: Here you can judge according to the `is_deleted` attribute of `db_class`, the default value is None

File: examples/softdelete.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+# and pass the Boolean field indicating whether a rule is deleted or not
+# your model needs to handle the update of fields
+# 'updated_by', 'updated_at', 'deleted_by', etc.
+adapter = Adapter(
+    engine,
+    CasbinSoftDeleteRule,
+    CasbinSoftDeleteRule.is_deleted,
+)
+# Create the Enforcer, etc.
+e = casbin.Enforcer("path/to/model.conf", adapter)
```
Comment (leeqvip) [inline examples/softdelete.py:89]: Please write it directly in the readme, in a separate section

**Review by leeqvip — CHANGES_REQUESTED** [review]

**Review by leeqvip — APPROVED** [review]

**Pull request merged**